lms/djangoapps/certificates/apis/v0/tests/test_views.py

```diff
@@ -0,0 +1,108 @@
+
+    def get_url(self, username):
+        """
+        Helper function to create the url for certificates
+        """
+        return reverse(
+            self.namespaced_url,
+            kwargs={
+                'course_id': self.course.id,
+                'username': username
+            }
+        )
+
+    def test_permissions(self):
+        """
+        Test that only the owner of the certificate can access the url
+        """
+        # anonymous user
+        resp = self.client.get(self.get_url(self.student.username))
+        self.assertEqual(resp.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        # another student
+        self.client.login(username=self.student_no_cert.username, password='test')
+        resp = self.client.get(self.get_url(self.student.username))
+        # gets 404 and 403 for security reasons
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(resp.data, {u'detail': u'Not found.'})  # pylint: disable=no-member
+        self.client.logout()
+
+        # same student of the certificate
+        self.client.login(username=self.student.username, password='test')
+        resp = self.client.get(self.get_url(self.student.username))
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+    def test_no_certificate_for_user(self):
+        """
+        Test for case with no certificate available
+        """
+        self.client.login(username=self.student_no_cert.username, password='test')
+        resp = self.client.get(self.get_url(self.student_no_cert.username))
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertIn('error_code', resp.data)  # pylint: disable=no-member
+        self.assertEqual(
+            'No certificate available for user "{}" for the course "{}"'.format(
+                self.student_no_cert.username,
+                self.course.id
+            ),
+            resp.data['error_code']  # pylint: disable=no-member
+        )
+
+    def test_certificate_for_user(self):
+        """
+        Tests case user that pulls her own certificate
+        """
+        self.client.login(username=self.student.username, password='test')
+        resp = self.client.get(self.get_url(self.student.username))
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(resp.data.get('username'), self.student.username)  # pylint: disable=no-member
+        self.assertEqual(resp.data.get('status'), CertificateStatuses.downloadable)  # pylint: disable=no-member
+        self.assertEqual(resp.data.get('grade'), '0.88')  # pylint: disable=no-member
+        self.assertEqual(resp.data.get('download_url'), 'www.google.com')  # pylint: disable=no-member
+        self.assertEqual(resp.data.get('certificate_type'), CourseMode.VERIFIED)  # pylint: disable=no-member
+        self.assertEqual(resp.data.get('course_id'), unicode(self.course.id))  # pylint: disable=no-member
```

Comment:
minor / nitpick: since this is a unit test I think it doesn't hurt to be more aggressive and use direct dictionary key access (e.g. `resp.data['username']`) instead of `.get()`.